### PR TITLE
Done :)

### PR DESCRIPTION
**Pull request by lonewolfdesignUK**
I have fully tested this, and I had a ton of fun making this! Was so satisfying 😃 

Thanks for making a beginner friendly issue! @badosu hope the code is okay.

### CODE REVIEW

**Comment by lonewolfdesignUK** [comment]
@WatchTheFort PR please and thank you :)

**Comment by badosu** [comment]
Looks good to me :+1: 

Good job!

File: luaui/Widgets/gui_camera_anchors.lua [hunk 1]
```diff
@@ -8,15 +8,16 @@ function widget:GetInfo()
 	return {
 		name      = "Camera Anchors",
 		desc      = "Adds keybindings for Camera Anchors",
-		author    = "badosu",
-		date      = "Oct 06, 2021",
+		author    = "badosu, lonewolfdesign",
+		date      = "Mar 12, 2023",
```
Comment (badosu) [inline luaui/Widgets/gui_camera_anchors.lua:12]: Not sure if the convention is date of creation, most drastic modification or last update but I really don't care to be honest.

File: luaui/Widgets/gui_camera_anchors.lua [hunk 2]
```diff
@@ -30,21 +31,20 @@ function SetCameraAnchor(_, _, args)
 	local anchorId = args[1]
 	local cameraState = GetCameraState()
```
Comment (badosu) [inline luaui/Widgets/gui_camera_anchors.lua:32]: Actually, can you confirm again that you are restoring the zoom level at the time of anchor set? `GetCameraState()` might differ with `GetCameraState(false)` but I'm not really used to camera callouts.

Comment (lonewolfdesignUK) [reply luaui/Widgets/gui_camera_anchors.lua:32]: Yeah when I switch anchors it changes the zoom but also rotation of the camera!

**Comment by lonewolfdesignUK** [comment]
> Looks good to me :+1: 
> 
> Good job!

Could I have another one? 😄

File: luaui/Widgets/gui_camera_anchors.lua (hunk 2 above)
Comment (badosu) [reply luaui/Widgets/gui_camera_anchors.lua:32]: Nice, glad to hear that :clap:

**Comment by badosu** [comment]
> > Looks good to me +1
> > Good job!
> 
> Could I have another one? smile

I'm afraid I don't really have any more easy issues (there might be one in a few days depending on a fellow contributors response). There is some stuff to improve that is moderately straightforward, from the top of my head:

- Invalidate building wind from gadgets instead of widgets according to certain criteria (consult with @WatchTheFort and/or @Ruwetuin to check what is the design decision on that). See also https://github.com/beyond-all-reason/Beyond-All-Reason/pull/1512
- Invalidate commander positioning on invalid terrain (steep slopes for example) during pregamestart via gadget instead of what I'm doing on build/gridmenu: https://github.com/beyond-all-reason/Beyond-All-Reason/issues/1309

For these you might require assistance from fellow developers on discord (or on matrix/IRC where I live).

**Comment by lonewolfdesignUK** [comment]
> > > Looks good to me +1
> > > Good job!
> > 
> > 
> > Could I have another one? smile
> 
> I'm afraid I don't really have any more easy issues (there might be one in a few days depending on a fellow contributors response). There is some stuff to improve that is moderately straightforward, from the top of my head:
> 
> * Invalidate building wind from gadgets instead of widgets according to certain criteria (consult with @WatchTheFort and/or @Ruwetuin to check what is the design decision on that). See also [fix(buildmenu): Unbuildable wind on rosetta #1512](https://github.com/beyond-all-reason/Beyond-All-Reason/pull/1512)
> * Invalidate commander positioning on invalid terrain (steep slopes for example) during pregamestart (startboxes): [Pregame/Placement: Commander can be placed on non-traversable terrain. #1309](https://github.com/beyond-all-reason/Beyond-All-Reason/issues/1309)
> 
> For these you might require assistance from fellow developers on discord (or on matrix/IRC where I live).

How do I get on the IRC? :)

**Comment by badosu** [comment]
[Here](https://web.libera.chat/#beyond-all-reason]

**Pull request merged**